packages/sanity/src/core/store/_legacy/kv/types.ts

```diff
@@ -0,0 +1,14 @@
+import {type Observable} from 'rxjs'
+
+type JsonObject = {[Key in string]: KVStoreValue} & {[Key in string]?: KVStoreValue | undefined}
+type JsonArray = KVStoreValue[] | readonly KVStoreValue[]
+type JsonPrimitive = string | number | boolean | null
+
+/** @internal */
+export type KVStoreValue = JsonPrimitive | JsonObject | JsonArray
```

Comment:
This JSON logic is taken from our `migrate` package. Should we generalize this so we can use it elsewhere?